ddsrouter_core/src/cpp/writer/implementations/rtps/filter/RepeaterDataFilter.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file RepeaterParticipant.hpp
+ */
+
+#ifndef __SRC_DDSROUTERCORE_PARTICIPANT_IMPLEMENTATIONS_RTPS_FILTER_REPEATERDATAFILTER_HPP_
+#define __SRC_DDSROUTERCORE_PARTICIPANT_IMPLEMENTATIONS_RTPS_FILTER_REPEATERDATAFILTER_HPP_
+
+#include <writer/implementations/rtps/filter/OriginDataFilter.hpp>
+
+namespace eprosima {
+namespace ddsrouter {
+namespace core {
+namespace rtps {
+
+/**
+ * This filter allows to not send messages from this Writer to the Readers belonging to the source Participant.
+ * This is used to "repeat" messages to all connected Readers but the one that has send it previously.
```

Comment:
It is used in "repeater" participants in order to propagate information to external participants (participants not belonging to the same DDS-Router instance), leaving out the participant from which this information was received.